driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -739,6 +758,25 @@ private boolean addConnectionIfUnderMaximum(int shardId, ConnectionTasksSharedSt
     }
   }
 
+  private int getAvailablePortForShard(int shardCount, int shardId, int maxAttempts) {
+    for (int i = 0; i < maxAttempts; i++) {
+      int port = 10000 + RAND.nextInt(50000 - shardCount);
+      port = port - port % shardCount + shardId;
```

Comment:
@jul-stas Please verify this is the right formula

Reply:
It's correct, but I would do it in a different way:
- 10000-60000 is not a safe range on some systems. This should be configurable.
- (minor nit) I would drop the randomness in favor of ordered search and a synchronized static method. It will be slower, but helps determining the order of socket binding.